### PR TITLE
Allow ulid to be nil

### PR DESCRIPTION
**Pull request by chaymaeBZ**
Given these 2 models :
```ruby
class User < ActiveRecord::Base
  include ULID::Rails
  ulid :id, primary_key: true

  has_many :books
end
```
```ruby
class Book < ActiveRecord::Base
  include ULID::Rails
  ulid :id, primary_key: true
  ulid :user_id

  belongs_to :user
end
```

if you do a `Book.create!` everything is cool because we are not explicitly assigning a ulid value (and looking at this book record, the `user_id` is `nil`). however, a `Book.create!(user: nil)` will fail because it doesn't pass the validation that expects a ulid instance to be of length 26.

### CODE REVIEW

**Comment by bquorning** [comment]
I don’t think I understand when you’d want to allow a `nil` ulid. Could you add a test case, please?

File: lib/ulid/rails/validator.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ module ULID
   module Rails
     module Validator
       def self.is_valid?(v)
+        return true if v.nil?
         v.length == 26 && Base32::Crockford.valid?(v)
```
Comment (bquorning) [inline lib/ulid/rails/validator.rb:8]:
```suggestion
        v && v.length == 26 && Base32::Crockford.valid?(v)
```

**Review by bquorning — APPROVED** [review]
Ahh, yes yes. If the user wants a presence check, it should be done in the model – not here, in the ulid library 👍🏼

File: test/ulid/rails_test.rb [hunk 2]
```diff
@@ -44,6 +44,12 @@ def test_validate_ulid_format
     assert_raises(ArgumentError) { User.new(id: non_ulid_id) }
   end
 
+  def test_should_save_when_ulid_foreign_key_is_nil
```
Comment (anitsirc) [inline test/ulid/rails_test.rb:47]: Avoid using "should" when describing your tests, express it with an action instead.

"should do this" -> "does this"

```suggestion
  def test_saves_when_ulid_foreign_key_is_nil
```

**Review by anitsirc — APPROVED** [review]
Very minor comment, else good catch 😊

File: lib/ulid/rails/validator.rb (hunk 1 above)
Comment (chaymaeBZ) [reply lib/ulid/rails/validator.rb]: `!v && v.length == 26 && Base32::Crockford.valid?(v)` :D

Comment (chaymaeBZ) [reply lib/ulid/rails/validator.rb:8]: when v is this eval to nil. which means false. which is not what we want. when v is nil, we want to return true (value is valid when v is nil)

**Pull request merged**